datafusion/sqllogictest/test_files/unnest.slt

```diff
@@ -511,10 +511,19 @@ x y [30, 40, 50]
 query error DataFusion error: type_coercion\ncaused by\nThis feature is not implemented: Unnest should be rewritten to LogicalPlan::Unnest before type coercion
 select sum(unnest(generate_series(1,10)));
 
-## TODO: support unnest as a child expr
 query error DataFusion error: Internal error: unnest on struct can only be applied at the root level of select expression
 select arrow_typeof(unnest(column5)) from unnest_table;
```

Comment:
This error isn't a child expression issue. It's an issue about applying a function to an unnest struct expression. The error message makes sense to me. DuckDB has similar behavior:
```sql
D CREATE TABLE unnest_table
  AS VALUES
      ([1,2,3], [7], 1, [13, 14], {'c1': 1, 'c2': 2}),
      ([4,5], [8,9,10], 2, [15, 16], {'c1':3,'c2':4}),
      ([6], [11,12], 3, null, null),
      ([12], [null, 42, null], null, null, {'c1':7,'c2':8}),
      -- null array to verify the `preserve_nulls` option
      (null, null, 4, [17, 18], null)
  ;
D select typeof(unnest(col4)) from unnest_table;
Binder Error: UNNEST() on a struct column can only be applied as the root element of a SELECT expression
LINE 1: select typeof(unnest(col4)) from unnest_table
```